src/components/StarRating/StarRating.jsx

```diff
@@ -0,0 +1,59 @@
+import { Component } from 'react';
+import * as S from './StarRating.styles';
+
+class StarRating extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      stars: [],
+      rating: 0,
+      hovered: 0,
+      selectedIcon: '★',
+      deselectedIcon: '☆'
+    };
+
+    let outOf = props.outOf ? props.outOf : 5;
+
+    for (let i = 0; i < outOf; i++) {
+      this.state.stars.push(i + 1);
```

Comment:
hint: you can try to use es6 array methods for generate array a bit shorter way

Reply:
Thank you, I have changed.